packages/components/src/Section/src/Section.tsx

```diff
@@ -0,0 +1,53 @@
+import { useStyledSystem, type StyledComponentProps } from "@hopper-ui/styled-system";
+import clsx from "clsx";
+import { type CSSProperties, forwardRef, type ForwardedRef } from "react";
+import { Section as RACSection, useContextProps, type SectionProps as RACSectionProps } from "react-aria-components";
+
+import { SectionContext } from "./SectionContext.ts";
+
+export const GlobalSectionCssSelector = "hop-Section";
+
+export interface SectionProps<T> extends StyledComponentProps<RACSectionProps<T>> {}
+
+function Section<T extends object>(props: SectionProps<T>, ref: ForwardedRef<HTMLElement>) {
+    [props, ref] = useContextProps(props, ref, SectionContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children,
+        style,
+        ...otherProps
+    } = ownProps;
+
+    const classNames = clsx(
+        GlobalSectionCssSelector,
+        stylingProps.className,
+        className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...stylingProps.style,
+        ...style
+    };
+
+    return (
+        <RACSection
+            {...otherProps}
+            ref={ref}
+            className={classNames}
+            style={mergedStyles}
+        >
+            {children}
+        </RACSection>
+    );
+}
+
+/**
+ * A Section serves as a versatile container component, encapsulating an HTML section element to facilitate the organization and grouping of content.
+ *
+ * [View Documentation](TODO)
+ */
+const _Section = forwardRef<HTMLElement, SectionProps<object>>(Section);
```

Comment:
`SectionProps<object>` makes it to the Section component is not really generic. 

i've seen something like this in spectrum:
 ```
const _Accordion = forwardRef(Accordion) as <T>(props: SpectrumAccordionProps<T> & {ref?: DOMRef<HTMLDivElement>}) => ReturnType<typeof Accordion>;

```